clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -227,4 +301,130 @@ private KeyStore load() throws GeneralSecurityException, IOException {
         }
     }
 
+    static class SSLConfigValidatorEngine {
+        private static final ByteBuffer EMPTY_BUF = ByteBuffer.allocate(0);
+        private final SSLEngine sslEngine;
```

Comment:
The `SSLEngine` documentation mentions that the engine should be closed after we have finished with it. It wasn't clear to me if that was needed here.

Reply:
Don't think it is needed here, but added anyway.